Include/internal/pycore_tstate.h

```diff
@@ -22,7 +22,7 @@ typedef struct _PyThreadStateImpl {
 
 #ifdef Py_GIL_DISABLED
     struct _mimalloc_thread_state mimalloc;
-    struct _Py_freelist_state freelist_state;
+    struct _PyObject_freelists freelist_state;
```

Comment:
`freelists` as the field name would be consistent with the per-interpreter definition.